### PR TITLE
Done DLPAR rework for dedicated and shared mode(new).

### PR DESCRIPTION
**Pull request by SamirMulani**
Below code cahages are done,

-> To execute the HMC commands in the old implementation, we used pexpect's python library for ssh operations. In the new implementation, we are using avocado's ssh library instead. -> The implementation code for the pexpect way of communicatinon has been removed from dlpar_api/api.py -> Created common API's for dedicated and shared mode CPU and memory,
	1. CPU, memory add, remove and move operation. -> Added new README.md
-> Removed dlpar config file(test.cfg) option to configure the DLPAR operations(dlpar_api/api.py). -> Removed workload implementation (dlpar_workload_setup.py).

### CODE REVIEW

**Comment by PraveenPenguin** [comment]
@SamirMulani run log please

File: dlpar/README.md [hunk 1]
```diff
@@ -13,68 +13,31 @@ OR
...
-#end - config/test.cfg
+# Sample dlpar.yaml is provided as below:
+#cat dlpar_main.py.data/dlpar.yaml
+
+cfg_cpu_per_proc: 8
+hmc_partition:   <partition name, for hmc it is same as managed system name>
+hmc_user:        <hmc default user, ex: hscroot>
+hmc_passwd:      <hscroot password>
+target_lpar_name: <fully qualified  lpar name>
+target_partition: <partition name, should be same as lpar name>
```
Comment (abdhaleegit) [inline dlpar/README.md:26]: just change the documentation 
target_lpar_hostname:
target_partition:

**Comment by SamirMulani** [comment]
Run logs for DLPAR add, move & remove operations for **memory** and **CPU**.

**Dedicated mode:**
```
[root@ltcden10-lp7 dlpar]# avocado run dlpar_main.py -m dlpar_main.py.data/dlpar.yaml --max-parallel-tasks=1
JOB ID     : 16f921ce597ae78d0809bd3a87bfa461feea5579
JOB LOG    : /Workspace/AV/avocado-fvt-wrapper/results/job-2023-03-13T03.09-16f921c/job.log
 (1/2) dlpar_main.py:DlparTests.test_cpu_dlpar;run-config-lpar_mode-dedicated-39dd: STARTED
 (1/2) dlpar_main.py:DlparTests.test_cpu_dlpar;run-config-lpar_mode-dedicated-39dd: PASS (456.23 s)
 (2/2) dlpar_main.py:DlparTests.test_mem_dlpar;run-config-lpar_mode-dedicated-39dd: STARTED
 (2/2) dlpar_main.py:DlparTests.test_mem_dlpar;run-config-lpar_mode-dedicated-39dd: PASS (192.71 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /Workspace/AV/avocado-fvt-wrapper/results/job-2023-03-13T03.09-16f921c/results.html
JOB TIME   : 650.92 s
```

**Shared Mode:**
```
[root@ltcden10-lp7 dlpar]# avocado run dlpar_main.py -m dlpar_main.py.data/dlpar.yaml --max-parallel-tasks=1
JOB ID     : f17e498cacfc07cdbdc944636c4475a192116aa7
JOB LOG    : /Workspace/AV/avocado-fvt-wrapper/results/job-2023-03-13T01.24-f17e498/job.log
 (1/2) dlpar_main.py:DlparTests.test_cpu_dlpar;run-config-lpar_mode-shared-c411: STARTED
 (1/2) dlpar_main.py:DlparTests.test_cpu_dlpar;run-config-lpar_mode-shared-c411: PASS (382.18 s)
 (2/2) dlpar_main.py:DlparTests.test_mem_dlpar;run-config-lpar_mode-shared-c411: STARTED
  (2/2) dlpar_main.py:DlparTests.test_mem_dlpar;run-config-lpar_mode-shared-c411: PASS (194.13 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /Workspace/AV/avocado-fvt-wrapper/results/job-2023-03-13T01.24-f17e498/results.html
JOB TIME   : 578.63 s
```

**Review by PraveenPenguin — APPROVED** [review]
LGTM

**Review by abdhaleegit — APPROVED** [review]
@SamirMulani Thanks a lot for working on cleanup and review comments.. we are good to merge

LGTM

**Pull request merged**